### PR TITLE
Adds backup and restore scripts

### PR DESCRIPTION
**Pull request by nickschuch**

### CODE REVIEW

File: containers/mysql/scripts/database-backup.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Name:        database-backup.sh
+# Description: Helper script for backing up a database.
+# Author:      Nick Schuch
+
+FILE=$1
+
+if [ "$DATABASE_HOST" == "" ]; then
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-backup.sh]:
```suggestion
if [ -z $DATABASE_HOST ]; then
```

File: containers/mysql/scripts/database-backup.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  echo "Not found: DATABASE_HOST"
+  exit 1
+fi
+
+if [ "$DATABASE_PORT" == "" ]; then
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-backup.sh]:
```suggestion
if [ -z $DATABASE_PORT ]; then
```

File: containers/mysql/scripts/database-backup.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  echo "Not found: DATABASE_PORT"
+  exit 1
+fi
+
+if [ "$DATABASE_USER" == "" ]; then
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-backup.sh]:
```suggestion
if [ -z $DATABASE_USER ]; then
```

File: containers/mysql/scripts/database-backup.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  echo "Not found: DATABASE_USER"
+  exit 1
+fi
+
+if [ "$DATABASE_PASSWORD" == "" ]; then
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-backup.sh]:
```suggestion
if [ -z $DATABASE_PASSWORD ]; then
```

File: containers/mysql/scripts/database-restore.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+if [ "$FILE" == "" ]; then
+  echo "Not found: FILE"
+  exit 1
+fi
+
+# Connection string which will be used to perform operations on the MySQL database.
+CONNECTION_STRING="mysql --host=$DATABASE_HOST --port=$DATABASE_PORT --user=$DATABASE_USER --password=$DATABASE_PASSWORD $DATABASE_NAME"
+
+TABLES=$($CONNECTION_STRING -e 'show tables' | $AWK '{ print $1}' | $GREP -v '^Tables' )
```
Comment (acbramley) [inline containers/mysql/scripts/database-restore.sh]: Missing $AWK and $GREP setters

File: containers/mysql/scripts/database-backup.sh [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  echo "Not found: DATABASE_PASSWORD"
+  exit 1
+fi
+
+if [ "$DATABASE_NAME" == "" ]; then
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-backup.sh]:
```suggestion
if [ -z $DATABASE_NAME ]; then
```

File: containers/mysql/scripts/database-backup.sh [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  echo "Not found: DATABASE_NAME"
+  exit 1
+fi
+
+if [ "$FILE" == "" ]; then
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-backup.sh]:
```suggestion
if [ -z $FILE ]; then
```

File: containers/mysql/scripts/database-restore.sh [hunk 8]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Name:        database-restore.sh
+# Description: Helper script for dropping tables and importing up a database.
+# Author:      Nick Schuch
+
+FILE=$1
+
+if [ "$DATABASE_HOST" == "" ]; then
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-restore.sh]: Same `-z` treatment to all these envvar checks.

File: containers/mysql/scripts/database-backup.sh [hunk 9]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-backup.sh]: Add `set -euo pipefail` below shebang.

File: containers/mysql/scripts/database-restore.sh [hunk 10]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-restore.sh]: Add `set -euo pipefail` below shebang.

File: containers/mysql/scripts/database-restore.sh [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+fi
+ 
+for table in $TABLES
+do
+	echo "Deleting $DATABASE_NAME/$table"
+	$CONNECTION_STRING -e "drop table $table"
+done
+
+echo "Importing database: $FILE"
+$CONNECTION_STRING < $FILE
```
Comment (nicksantamaria) [inline containers/mysql/scripts/database-restore.sh]: Add a `echo "Import complete"` message at bottom to confirm mysql client didnt error out.

**Comment by nicksantamaria** [comment]
Yeah buddy!

sealofapproval

**Pull request merged**